Directory.Build.props

```diff
@@ -4,29 +4,28 @@
      Adding Sdk="Microsoft.NET.Sdk" to a project causes NuGet files to be generated very early on, so we need these defines to be here
      to ensure that all of our build output files end up in the same location.  This is parsed too late when put in mux.controls.props. -->
 <Project InitialTargets="DirectoryBuildPropsInfo" ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <PropertyGroup Condition="$(BuildingWithBuildExe) != 'true'">
-    <IsTDPConfiguration>false</IsTDPConfiguration>
-
-    <BaseOutputPath>$(MSBuildThisFileDirectory)BuildOutput\$(Configuration)</BaseOutputPath>
-    <OutDir>$(BaseOutputPath)\$(Platform)\</OutDir>
 
-    <!-- For native projects that use Win32 to mean x86, override the OutDir
-         property to provide consistency between projects. -->
-    <OutDir Condition="'$(Platform)' == 'Win32'">$(BaseOutputPath)\x86\</OutDir>
+  <!-- Due to import ordering inconsistencies between different project types, we import this file from multiple places. -->
+  <Import Project="$(MSBuildThisFileDirectory)build\EnsureOutputLayout.props" Condition="$(EnsureOutputLayoutPropsIncluded) == ''" />
 
-    <OutputPath>$(OutDir)\$(MSBuildProjectName)\</OutputPath>
+  <PropertyGroup>    
+    <IsTDPConfiguration>false</IsTDPConfiguration>
 
     <GeneratedFilesDir>$(BaseIntermediateOutputPath)Generated Files\</GeneratedFilesDir>
     <GenerateProjectSpecificOutputFolder>True</GenerateProjectSpecificOutputFolder>
+
+    <CustomBeforeMicrosoftCommonTargets>$(BuildRoot)BeforeCommon.targets</CustomBeforeMicrosoftCommonTargets>
+    <ForceImportAfterCppTargets>$(BuildRoot)AfterCpp.targets</ForceImportAfterCppTargets>
   </PropertyGroup>
 
   <!-- Compilation differs for the build pipeline vs local development -->
   <PropertyGroup Condition="$(ProjectReunionBuildPipeline) != '1'">
     <!-- If not set, default it to no . -->
     <ProjectReunionBuildPipeline>0</ProjectReunionBuildPipeline>
   </PropertyGroup>
-  <Target Name="DirectoryBuildPropsInfo" Condition="$(ProjectReunionBuildPipeline) == '1'">
-    <Message Importance="High" Text="Directory.Build.props detects ProjectReunionBuildPipeline=$(ProjectReunionBuildPipeline)"/>
+
+  <Target Name="DirectoryBuildPropsInfo">
+    <Message Condition="$(ProjectReunionBuildPipeline) == '1'" Importance="High" Text="Directory.Build.props detects ProjectReunionBuildPipeline=$(ProjectReunionBuildPipeline)"/>
```

Comment:
>$(ProjectReunionBuildPipeline) [](http://example.com/codeflow?start=145&length=30)

print "ProjectReunionBuildPipeline=1" doesn't make too much sense to most people. Maybe a more human readable "detect running in build pipeline"?

Reply:
ProjectReunionBuildPipeline is an optional build parameter, and this block is a build debugging tool. I'd rather see "foo=bar" with the actual value than "foo is in use" or some such English prose.

This was also done to avoid assuming there'd be no other values in the future. =2, =foo, whatever. Depending how the build pipeline evolves, but not precluding it, and not making it more difficult than necessary if/when we choose to go there.

Reply:
i'd agree if it always printed $(ProjectReunionBuildPipeline). However, here it only prints 1.

Reply:
I originally had it w/o the condition, which is fine for pipeline builds but it also caused spew in inner-loop builds The condition was put in (on the parent node) to avoid that noise

And it always prints $(ProjectReunionBuildPipeline), it just so happens that only has a value of 1 today. As I said, left open for other uses as the build pipelines future was very uncertain (and is still fuzzy)